pkg/conda/meta/parse_test.go

```diff
@@ -0,0 +1,142 @@
+		},
+		{
+			name:  "libgcc",
+			input: "testdata/libgcc-ng-11.2.0-h1234567_1.json",
+			want:  []types.Library{{Name: "libgcc-ng", Version: "11.2.0", License: "GPL-3.0-only WITH GCC-exception-3.1"}},
+		},
+		{
+			name:  "libgomp",
+			input: "testdata/libgomp-11.2.0-h1234567_1.json",
+			want:  []types.Library{{Name: "libgomp", Version: "11.2.0", License: "GPL-3.0-only WITH GCC-exception-3.1"}},
+		},
+		{
+			name:  "libstdcxx",
+			input: "testdata/libstdcxx-ng-11.2.0-h1234567_1.json",
+			want:  []types.Library{{Name: "libstdcxx-ng", Version: "11.2.0", License: "GPL-3.0-only WITH GCC-exception-3.1"}},
+		},
+		{
+			name:  "ncurses",
+			input: "testdata/ncurses-6.3-h5eee18b_3.json",
+			want:  []types.Library{{Name: "ncurses", Version: "6.3", License: "Free software (MIT-like)"}},
+		},
+		{
+			name:  "openssl",
+			input: "testdata/openssl-1.1.1q-h7f8727e_0.json",
+			want:  []types.Library{{Name: "openssl", Version: "1.1.1q", License: "OpenSSL"}},
+		},
+		{
+			name:  "pip",
+			input: "testdata/pip-22.2.2-py38h06a4308_0.json",
+			want:  []types.Library{{Name: "pip", Version: "22.2.2", License: "MIT"}},
+		},
+		{
+			name:  "python",
+			input: "testdata/python-3.8.8-hdb3f193_5.json",
+			want:  []types.Library{{Name: "python", Version: "3.8.8", License: "Python-2.0"}},
+		},
+		{
+			name:  "readline",
+			input: "testdata/readline-8.2-h5eee18b_0.json",
+			want:  []types.Library{{Name: "readline", Version: "8.2", License: "GPL-3.0-only"}},
+		},
+		{
+			name:  "setuptools",
+			input: "testdata/setuptools-65.5.0-py38h06a4308_0.json",
+			want:  []types.Library{{Name: "setuptools", Version: "65.5.0", License: "MIT"}},
+		},
+		{
+			name:  "sqlite",
+			input: "testdata/sqlite-3.39.3-h5082296_0.json",
+			want:  []types.Library{{Name: "sqlite", Version: "3.39.3", License: "blessing"}},
+		},
+		{
+			name:  "tk",
+			input: "testdata/tk-8.6.12-h1ccaba5_0.json",
+			want:  []types.Library{{Name: "tk", Version: "8.6.12", License: "TCL"}},
+		},
+		{
+			name:  "wheel",
+			input: "testdata/wheel-0.37.1-pyhd3eb1b0_0.json",
+			want:  []types.Library{{Name: "wheel", Version: "0.37.1", License: "MIT"}},
+		},
+		{
+			name:  "xz",
+			input: "testdata/xz-5.2.6-h5eee18b_0.json",
+			want:  []types.Library{{Name: "xz", Version: "5.2.6", License: "LGPL-2.1-or-later and GPL-2.0-or-later"}},
+		},
+		{
+			name:  "zlib",
+			input: "testdata/zlib-1.2.13-h5eee18b_0.json",
+			want:  []types.Library{{Name: "zlib", Version: "1.2.13", License: "Zlib"}},
+		},
+		{
+			name:    "invalid",
+			input:   "testdata/invalid.json",
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f, err := os.Open(tt.input)
+			require.NoError(t, err)
+
+			got, _, err := meta.NewParser().Parse(f)
+			var errMsg string
+			if err != nil {
+				errMsg = err.Error()
+			}
+			require.Equal(t, tt.wantErr, err != nil, errMsg)
+
+			assert.Equal(t, tt.want, got)
```

Comment:
I think it's better to use this logic
https://github.com/aquasecurity/go-dep-parser/blob/e0bd11321acbfe403ae0ecf502fe7488ce6dbfa6/pkg/rust/binary/parse_test.go#L79-L87

Reply:
Updated the test logic